### PR TITLE
Unit Tests

### PR DESCRIPTION
**Pull request by 69mb**
All model classes tested.

Utility classes pending and can be done going forward.

### CODE REVIEW

File: application/tests/setup_db.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	echo "Usage: $0 {Install-Script} {MySQL-Host-Name} {MySQL-Database-Name} {MySQL-User-Name} {MySQL-User-Password}"
+	echo "Drops all tables from a MySQL"
+	exit 1
+fi
+
+MYSQL_OPTIONS="-h $MYSQL_HOST -u $MYSQL_USER $MYSQL_DB"
+
+if [ $MYSQL_PASS ]; then
+	echo "HERE"
+	exit
```
Comment (ekala) [inline application/tests/setup_db.sh]: If the user for the test database has a password, the script will exit so the schema setup script will not be executed

**Comment by 69mb** [comment]
Removed the debug stuff.

**Comment by ekala** [comment]
+1

Going forward, the tests should also have the `@covers` doc tag. This is useful for code coverage analysis.

**Comment by 69mb** [comment]
Merged but didn't put coverage intentionally since it is not really a useful metric.

**Pull request merged**